### PR TITLE
Enable `noUncheckedIndexedAccess`

### PR DESCRIPTION
**Pull request by wojpawlik**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Documentation (typos, code examples or any documentation update)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Node.js Version:
* Operating System:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/composer.ts
Comment (KnorpelSenf) [inline src/composer.ts:776]: Why would `next` need to be passed to `Composer.unwrap`? It won't be `Middleware.Obj`, will it?

Comment (wojpawlik) [reply src/composer.ts:776]: It won't be `Middleware.Obj`, but it actually happens to be assignable to `Middleware.Fn`.

It's passed to `Composer.unwrap` because `middlewares[i]` (now typed as possibly `undefined`) is.

Comment (KnorpelSenf) [reply src/composer.ts:776]: So what's wrong with
```suggestion
        const handler = middlewares[i] === undefined ? next : Composer.unwrap(middlewares[i])
```
?

Comment (wojpawlik) [reply src/composer.ts:776]: TypeScript is not smart enough to be able to narrow one element of array, I'd need to introduce a `const` for `middleares[i]`

Comment (KnorpelSenf) [reply src/composer.ts:776]: I find two lines of code still better than doing that function call for `next`. That's confusing because it's not necessary.

I know that this is a nitpick ;)

**Comment by wojpawlik** [comment]
I've just ran `npm ci`, and have no typing / linting issues on my machine... interesting...

**Review by KnorpelSenf — APPROVED** [review]
I'd still be happy to have https://github.com/telegraf/telegraf/pull/1239#discussion_r541912115 inserted before merging, but I won't let this block the PR.

File: src/composer.ts
Comment (wojpawlik) [reply src/composer.ts:776]: I found a better way to write this:

```ts
        const handler = middlewares[i]
        if (handler === undefined) return next()
        await Composer.unwrap(handler)(context, async (ctx = context) => {
          await execute(i + 1, ctx)
        })
```

**Pull request merged**
